frontend/src/pages/offers/CreateOfferForm.tsx

```diff
@@ -29,88 +31,126 @@ interface Props {
+        // Otherwise, display all the groups captained by that user
+        getGroups({
+          variables: {
+            groupType: [GroupType.SendingGroup],
+            captainId: profile.isAdmin ? undefined : profile.id,
+          },
+        })
+      }
+    },
+    [profile, getGroups],
+  )
+
   const {
     register,
     handleSubmit,
+    watch,
+    reset,
     formState: { errors },
   } = useForm<OfferCreateInput>()
+  const watchSendingGroupId = watch('sendingGroupId')
+
+  useEffect(
+    function updateContactInformation() {
+      if (groups?.listGroups) {
+        // When the sendingGroupId changes, we pre-fill the contact info
+        const sendingGroupId = watchSendingGroupId || profile?.groupId
+        const matchingGroup = groups.listGroups.find(
+          (group) => group.id === sendingGroupId,
+        )
+
+        if (matchingGroup != null) {
+          reset({
+            sendingGroupId,
+            contact: {
+              ..._pick(matchingGroup.primaryContact, [
+                'name',
+                'email',
+                'phone',
+                'signal',
+                'whatsApp',
+              ]),
+            },
+          })
+        }
+      }
+    },
+    [watchSendingGroupId, groups, reset],
+  )
 
   const onSubmitForm = (input: OfferCreateInput) => {
     if (!props.shipmentId) {
       throw new Error('The shipment ID is missing in the offer form')
     }
 
-    if (!groupForUser) {
-      throw new Error('Cannot find a shipping group in the offer form')
+    if (!groups) {
+      throw new Error('Unable to fetch groups')
     }
 
-    // The sending group ID and shipment ID aren't in the form, so we add them
-    // manually
-    input.shipmentId = props.shipmentId
-    input.sendingGroupId = groupForUser.id
-
-    // The backend doesn't want null values for optional fields
-    if (input.contact) {
-      input.contact = setEmptyFieldsToUndefined(input.contact)
+    const payload: OfferCreateInput = {
+      shipmentId: props.shipmentId,
+      sendingGroupId: input.shipmentId || groups.listGroups[0].id,
+      contact: setEmptyFieldsToUndefined(input.contact),
+      photoUris: input.photoUris || [],
     }
 
-    input.photoUris = input.photoUris || []
-
-    props.onSubmit(input)
+    props.onSubmit(payload)
   }
 
-  // TODO this won't work if the user is an admin
-  if (!profile?.isAdmin) {
-    if (!groupForUser || !profile) {
-      return (
-        <div>
-          <Spinner /> Loading form data
-        </div>
-      )
-    }
+  if (isLoadingGroups || !profile || !groups) {
+    return (
+      <div>
+        <Spinner /> Loading form data
+      </div>
+    )
   }
 
   return (
     <form onSubmit={handleSubmit(onSubmitForm)} className="space-y-6">
-      <ReadOnlyField label="Sending group">
-        {groupForUser?.name || 'No group'}
-      </ReadOnlyField>
+      {groups.listGroups.length > 1 ? (
```

Comment:
If the user has access to more than 1 group, we display a dropdown. Otherwise, we select the group for them.